.github/workflows/test.yml

```diff
@@ -8,24 +8,32 @@ on:
       - 'releases/*'
 
 jobs:
-  build: # make sure build/ci work properly
+  test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
       - name: Read .nvmrc
         id: nvm
         run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+
       - name: Setup Node.js
         uses: actions/setup-node@v1
         with:
           node-version: ${{ steps.nvm.outputs.NVMRC }}
-      - run: npm ci
-      - run: npm run all
+
+      - name: Install NPM dependencies
+        run: npm ci
+
+      - name: Check formatting
+        run: npm run format-check
+
+      - name: Run linter
+        run: npm run lint-check
+
+      - name: Run tests
+        run: npm run test
+
       - run: git diff --quiet dist/
-  test: # make sure the action works on a clean machine without building
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: ./
-        with:
-          milliseconds: 1000
```

Comment:
This now just lives in `check-dist` as it makes sense to do all verification there and focus this workflow on normal CI flow